common/src/main/scala/uk/ac/wellcome/finatra/modules/ElasticClientModule.scala

```diff
@@ -3,68 +3,43 @@ package uk.ac.wellcome.finatra.modules
 import javax.inject.Singleton
 
 import com.google.inject.Provides
-import com.sksamuel.elastic4s.xpack.security.XPackElasticClient
-import com.sksamuel.elastic4s.{ElasticsearchClientUri, TcpClient}
+import org.elasticsearch.client.RestClient
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback
+import com.sksamuel.elastic4s.http.HttpClient
 import com.twitter.inject.TwitterModule
-import org.elasticsearch.common.settings.Settings
-
-import scala.collection.JavaConverters._
-
-case class XPackConfig(user: String, ssl: Boolean)
-
-object XPackConfigModule extends TwitterModule {
-  private val xpackEnabled =
-    flag[Boolean]("es.xpack.enabled", false, "xpack enabled")
-  private val user = flag[String]("es.xpack.user", "", "xpack user:password")
-  private val sslEnabled =
-    flag[Boolean]("es.xpack.sslEnabled", true, "xpack use ssl")
-
-  @Singleton
-  @Provides
-  def providesXPackConfig(): Option[XPackConfig] =
-    Option(xpackEnabled()).collect {
-      case true =>
-        XPackConfig(user(), sslEnabled())
-    }
+import org.apache.http.auth.{AuthScope, UsernamePasswordCredentials}
+import org.apache.http.HttpHost
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder
+
+class ElasticCredentials (username: String, password: String) extends TwitterModule with HttpClientConfigCallback {
```

Comment:
No, good spot – an earlier version of this code defined the flags in this class, but then I moved them down to `ElasticCredentials`. I’ll remove it.